sys/net/gnrc/link_layer/netdev2/gnrc_netdev2_eth.c

```diff
@@ -0,0 +1,195 @@
+#include "net/gnrc.h"
+#include "net/gnrc/gnrc_netdev2.h"
+#include "net/ethernet/hdr.h"
+
+#include "od.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+static gnrc_pktsnip_t *_recv(gnrc_netdev2_t *gnrc_netdev2)
```

Comment:
Well, to get it work, I create a big-enough pktsnip, and then build another one, copying the data I need.

I tried to just mark out the ethernet header, snip_remove that, and than use the initial big pktsnip, add the netif hdr, go... but It didn't manage to get it to work without memory leaks.